### DISCUSSION TITLE
"Grub failed to install to the following devices do you want to continue anyway? If you do your computer may not start up properly."

### DISCUSSION THREAD

**Discussion by cpoptic** [microsoft/WSL#5910] [General]
I suspect this issue is related to having both Windows/Ubuntu-via-WSL 2 running (maybe?):

I'm on a Windows 10 machine that has WSL 2 installed (Ubuntu 20.04). I was updating/reinstalling Docker per instructions [(https://docs.docker.com/engine/install/ubuntu/)] through the Ubuntu terminal and after the sudo apt update && sudo apt upgrade command, was prompted with the following:

> The grub-pc package is being upgraded. This menu allows you to select which devices you'd like grub-install to be automatically run for, if any.
> 
> Running grub-install automatically is recommended in most situations, to prevent the installed GRUB core image from getting out of sync with GRUB modules or grub.cfg.
> 
> If you're unsure which drive is designated as boot drive by your BIOS, it is often a good idea to install GRUB to all of them.
> 
> Note: it is possible to install GRUB to partition boot records as well, and some appropriate partitions are offered here. However, this forces GRUB to use the blocklist mechanism, which makes it less reliable, and therefore is not recommended.
> 
> GRUB install devices:
> 
> [*] /dev/sda (274877 MB; ???)
> 
> [*] /dev/sdb (274877 MB; ???)
> 
> [*] /dev/sdb (274877 MB; ???)
> 
> [*] /dev/sdd (274877 MB; ???)
> 
> [*] /dev/sdd (274877 MB; ???)

I follow the suggestions and select all 5 devices, but then am blocked by this message:

> Grub failed to install to the following devices do you want to continue anyway? If you do your computer may not start up properly.

That last sentence has me terrified. What to do here?

# Environment

```none
Windows build number: Microsoft Windows [Version 10.0.20206.1000]
Your Distribution version:   20.04
Whether the issue is on WSL 2 and/or WSL 1:  I'm using WSL 2:   
> Linux version 4.19.104-microsoft-standard (oe-user@oe-host) (gcc version 8.2.0 (GCC)) #1 SMP Wed Feb 19 06:37:35 UTC 2020

```

# Steps to reproduce

**WSL logs**: 

#  Expected behavior

# Actual behavior

**Comment by cpoptic** [comment]
Okay, thanks that's what I suspected; as WSL2 is decoupled from the
physical metal, but I wanted to double check before doing anything
potentially irreversible.  Thanks so much.

**Comment by therealkenc** [comment]
There are a few related tickets eg #4903. There is probably a way to make the upstream grub package more tolerant of environments where is ends up installed but is not applicable, like WSL, Docker, diskless boots, etc. I tend to `apt remove` it when it gets installed as a side-effect of other packages. I suspect everyone else does too, which is probably why it has never been made more tolerant.
